File: examples/smart_contracts/v2/python/compile_teal.py
# compile TEAL code

from algosdk.v2client import algod
import os

# read file
def load_resource(res):
    dir_path = os.path.dirname(os.path.realpath(__file__))
    path = os.path.join(dir_path, res)
    with open(path, "rb") as fin:
        data = fin.read()
    return data


try:
    # create an algod client
    algod_token = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
    algod_address = "http://localhost:4001"
    algod_client = algod.AlgodClient(algod_token, algod_address)

    # int 1 - sample.teal
    # This code is meant for learning purposes only
    # It should not be used in production
    myprogram = "samplearg.teal"

    # read TEAL program
    data = load_resource(myprogram)
    source = data.decode("utf-8")

    # Compile TEAL program
    response = algod_client.compile(source)

    print("Response Result = ", response["result"])
    print("Response Hash = ", response["hash"])
except Exception as e:
    print(e)

# results should be the following:
# Response Result =  Bi0XgXsS
# Response Hash =  PBS3JWRY5HGL46SDYLDVMEJVJXKMSCFOAY7SKIW2RQ7OTJVWNNGW4QXQ5A

File: examples/smart_contracts/v2/javascript/compileTeal.js
const algosdk = require('algosdk');

const token = "a".repeat(64);
const server = "http://localhost";
const port = 4001;

// Import the filesystem module 
const fs = require('fs');

let algodclient = new algosdk.Algodv2(token, server, port);

(async () => {
    // Read file for Teal code - int 1
    var fs = require('fs'),
        path = require('path'),
        filePath = path.join(__dirname, 'samplearg.teal');
    let data = fs.readFileSync(filePath);
    let results = await algodclient.compile(data).do();
    console.log("Hash = " + results.hash);
    console.log("Result = " + results.result);

})().catch(e => {
    console.log(e.body.message);
    console.log(e);
});

// results should be the following:
// Response Result =  Bi0XgXsS
// Response Hash =  PBS3JWRY5HGL46SDYLDVMEJVJXKMSCFOAY7SKIW2RQ7OTJVWNNGW4QXQ5A